backend/src/utils/error.rs

```diff
@@ -13,56 +15,43 @@ pub enum Error {
     Request(#[from] reqwest::Error),
     #[error("Database operation failed")]
     DatabaseOperation(#[from] sea_orm::DbErr),
-    #[error("User already exists")]
-    UserAlreadyExists,
-    #[error("Invalid username and/or password")]
-    InvalidCredentials,
-    #[error("Invalid email address")]
-    InvalidEmailAddress,
-    #[error("Password & email authentication is not available for this account")]
-    PasswordAuthNotAvailable,
+    #[error("Unauthorized")]
+    Unauthorized,
     #[error("Invalid Json Web Token")]
     InvalidJsonWebToken,
     #[error("Invalid authorization header content")]
     InvalidAuthorizationHeader,
-    #[error("Unauthorized")]
-    Unauthorized,
-    #[error("Team already exists")]
-    TeamAlreadyExists,
-    #[error("User already belongs to team: {team_name}")]
-    UserAlreadyBelongsToTeam { team_name: String },
-    #[error("Team not found")]
-    TeamNotFound,
-    #[error("User doesn't belong to any team")]
-    UserDoesntBelongToAnyTeam,
+    #[error(transparent)]
+    Auth(#[from] AuthError),
+    #[error(transparent)]
+    Team(#[from] TeamError),
+}
+
+// Helper function to generate consistent JSON error responses
```

Comment:
This comment is unneccesary, but if you really want it, make it doc comment `///`